### PR TITLE
Enhancing sync schema for layout user syncs

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: shared/syncs.schema.json [hunk 1]
```diff
@@ -19,9 +19,18 @@
         "sobjectTypePlural": {
             "type": "string"
         },
+        "formFactor": {
```
Comment (wmathurin) [inline shared/syncs.schema.json]: We could go further and do:
```json
"formFactor": {
 "type": "string",
 "enum": ["Large", "Medium", "Small"]
}
```

File: shared/syncs.schema.json [hunk 2]
```diff
@@ -19,9 +19,18 @@
...
+            "type": "string"
+        },
         "layoutType": {
```
Comment (wmathurin) [inline shared/syncs.schema.json]: Same thing for layout type, we could add enum with the supported values. See https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_resources_record_layout.htm

File: shared/syncs.schema.json [hunk 3]
```diff
@@ -19,9 +19,18 @@
...
             "type": "string"
         },
+        "mode": {
```
Comment (wmathurin) [inline shared/syncs.schema.json]: Same thing for mode, we could add enum with the supported values. See https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_resources_record_layout.htm

**Comment by bhariharan** [comment]
Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-Package/pull/199/commits/d4e40010f67fc1398a0635d7ad99b4e7bc8900fb).

**Pull request merged**